src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -10,24 +10,24 @@
 /**
  * Base Global Styles REST API Controller.
  */
-class WP_REST_Global_Styles_Controller extends WP_REST_Controller {
-
+class WP_REST_Global_Styles_Controller extends WP_REST_Posts_Controller {
 	/**
-	 * Post type.
+	 * Whether the controller supports batching.
 	 *
-	 * @since 5.9.0
-	 * @var string
+	 * @since 6.5.0
+	 * @var array
 	 */
-	protected $post_type;
+	protected $allow_batch = array( 'v1' => false );
 
 	/**
 	 * Constructor.
-	 * @since 5.9.0
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $post_type Post type.
 	 */
-	public function __construct() {
-		$this->namespace = 'wp/v2';
-		$this->rest_base = 'global-styles';
-		$this->post_type = 'wp_global_styles';
+	public function __construct( $post_type = 'wp_global_styles' ) {
```

Comment:
This LGTM. I've tested with the plugin activated (with and without this change).

It also provides backwards compatibility for folks running older versions of Gutenberg. 

If folks think it's appropriate I can update the class instantiation in Gutenberg to pass `'wp_global_styles'` as well, to make things more consistent. But this PR's changes make that unnecessary I suppose.

https://github.com/WordPress/gutenberg/blob/b35bc15f13e2287befaa638a3a6aa168c31b5362/lib/compat/wordpress-6.5/rest-api.php#L17-L17